packages/fixtures/src/serverless-tester.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  APIGatewayProxyEvent,
+  APIGatewayProxyResult,
+  Context,
+} from 'aws-lambda';
+import crypto from 'crypto';
+import serverless, { Application, Handler } from 'serverless-http';
+
+export default class ServerlessTester {
+  private handler: Handler;
+
+  private readonly signingSecret: string;
+
+  constructor(app: Application, signingSecret: string) {
+    this.handler = serverless(app);
+    this.signingSecret = signingSecret;
+  }
+
+  private createSlackRequest(
+    data: any,
+    path = '/slack/events',
+  ): Partial<APIGatewayProxyEvent> {
+    const body = JSON.stringify(data);
+    const version = 'v0';
+    const timestamp = Math.floor(Date.now() / 1000);
+    const hmac = crypto.createHmac('sha256', this.signingSecret);
+
+    hmac.update(`${version}:${timestamp}:${body}`);
+
+    return {
+      body,
+      path,
+      headers: {
+        'content-type': 'application/json',
+        'x-slack-request-timestamp': timestamp.toString(),
+        'x-slack-signature': `${version}=${hmac.digest('hex')}`,
+      },
+      httpMethod: 'POST',
+    };
```

Comment:
had a thought that enabling one to build the request body separately from the whole request, and enable later providing a path/actually trigger the request might be useful, but can be a future thing.

Reply:
I'm intrigued and not following, pseudo code example?

Reply:
roughly exposing and currying this function.

so you could call `createSlackRequest(data)`, `createSlackRequest(data, path)`, or `createSlackRequest(data)(path)`, where it leaves/adds path to the request when given it. then, making it public so that people can create the request, do things with it themselves, and then do something like

```ts
pipe(
  createSlackRequest,
  transformRequest,
  sendRequest
)(data, path)
```
but anyway, not critical at the moment